app/services/induction/set_school_cohort_appropriate_body.rb

```diff
@@ -2,27 +2,35 @@
 
 class Induction::SetSchoolCohortAppropriateBody < BaseService
   def call
-    set_appropriate_body
-    school_cohort.save!
+    ActiveRecord::Base.transaction do
+      set_school_cohort_appropriate_body
+      set_induction_records_appropriate_body if update_induction_records
+    end
   end
 
 private
 
-  attr_reader :school_cohort, :appropriate_body_id, :appropriate_body_appointed
+  attr_reader :school_cohort, :appropriate_body_id, :appropriate_body_appointed, :update_induction_records
 
-  def initialize(school_cohort:, appropriate_body_id:, appropriate_body_appointed:)
+  def initialize(school_cohort:, appropriate_body_id:, appropriate_body_appointed:, update_induction_records: false)
     @school_cohort = school_cohort
     @appropriate_body_id = appropriate_body_id
     @appropriate_body_appointed = appropriate_body_appointed
+    @update_induction_records = update_induction_records
   end
 
-  def set_appropriate_body
-    if !appropriate_body_appointed
+  def set_school_cohort_appropriate_body
+    if appropriate_body_appointed == false
```

Comment:
Is there a reason for the `== false` here, rather than just doing `if appropriate_body_appointed` and inverting the 'then/else'? I guess it's explicitly guarding against nils?

Reply:
the appropriate_body_appointed field is used when the journey ask the user whether the appropriate body has been appointed or not.
There are some journeys where the user just chooses an appropriate body.
When the journey requires to skip the appropriate body appointed step then the value is nil (never used).
We need to differentiate when the value is explicitly set to false that's when the user explicitly state they don't know the appropriate body.
To answer your question, yes it is guarding against nil.

Reply:
@peteryates is that ok to leave it like this?
Otherwise I'm open to suggestions.